src/transformers/models/mask_rcnn/modeling_maskrcnn.py

```diff
@@ -0,0 +1,3818 @@
+        )
+    if pred.size(0) != target.size(0):
+        raise ValueError(f"pred and target should be of same size, but got {pred.size(0)} and {target.size(0)}")
+    if maxk > pred.size(1):
+        raise ValueError(f"maxk {maxk} exceeds pred dimension {pred.size(1)}")
+    pred_value, pred_label = pred.topk(maxk, dim=1)
+    pred_label = pred_label.t()  # transpose to shape (maxk, N)
+    correct = pred_label.eq(target.view(1, -1).expand_as(pred_label))
+    if thresh is not None:
+        # Only prediction values larger than thresh are counted as correct
+        correct = correct & (pred_value > thresh).t()
+    res = []
+    for k in topk:
+        correct_k = correct[:k].reshape(-1).float().sum(0, keepdim=True)
+        res.append(correct_k.mul_(100.0 / pred.size(0)))
+    return res[0] if return_single else res
+
+
+def cross_entropy(
+    pred,
+    label,
+    weight=None,
+    reduction="mean",
+    avg_factor=None,
+    class_weight=None,
+    ignore_index=-100,
+    avg_non_ignore=False,
+):
+    """Calculate the CrossEntropy loss.
+
+    Args:
+        pred (`torch.Tensor`):
+            The prediction with shape (N, C), C is the number of classes.
+        label (`torch.Tensor`):
+            The learning label of the prediction.
+        weight (`torch.Tensor`, *optional*):
+            Sample-wise loss weight.
+        reduction (`str`, *optional*, defaults to `"mean"`):
+            The method used to reduce the loss.
+        avg_factor (`int`, *optional*):
+            Average factor that is used to average the loss.
+        class_weight (`List[float]`, *optional*):
+            The weight for each class.
+        ignore_index (`int`, *optional*, defaults to -100):
+            The label index to be ignored.
+        avg_non_ignore (`bool`, *optional*, defaults to `False`):
+            The flag decides to whether the loss is only averaged over non-ignored targets.
+
+    Returns:
+        `torch.Tensor`: The calculated loss
+    """
+    # The default value of ignore_index is the same as nn.functional.cross_entropy
+    ignore_index = -100 if ignore_index is None else ignore_index
+    # element-wise losses
+    loss = nn.functional.cross_entropy(pred, label, weight=class_weight, reduction="none", ignore_index=ignore_index)
+
+    # average loss over non-ignored elements
+    # pytorch's official cross_entropy average loss over non-ignored elements
+    # refer to https://github.com/pytorch/pytorch/blob/56b43f4fec1f76953f15a627694d4bba34588969/torch/nn/functional.py#L2660  # noqa
+    if (avg_factor is None) and avg_non_ignore and reduction == "mean":
+        avg_factor = label.numel() - (label == ignore_index).sum().item()
+
+    # apply weights and do the reduction
+    if weight is not None:
+        weight = weight.float()
+    loss = weight_reduce_loss(loss, weight=weight, reduction=reduction, avg_factor=avg_factor)
```

Comment:
In stead of having this scale factor. We can:  
* Set `weight` to a vector of ones of size labels by default 
* mask out the weight, pred and label where label == ignore_index 
* pass those values to `nn.functional.cross_entropy`
* Average with the sample weights. 

This will reduce some of the if/else logic and make it clearer what's being averaged.